third_party/move/move-compiler-v2/tests/checking/typing/recursive_struct.exp

```diff
@@ -0,0 +1,122 @@
+
+Diagnostics:
+error: recursive definition S
+  ┌─ tests/checking/typing/recursive_struct.move:2:2
+  │
+2 │ ╭     struct S {
+3 │ │         f: T
+4 │ │     }
+  │ ╰─────^
+  │
+  = `simple_recursion::S` contains field `f: simple_recursion::T`...
+  = `simple_recursion::T` contains field `f: S`, which forms a loop.
+
+error: recursive definition T
+  ┌─ tests/checking/typing/recursive_struct.move:5:2
+  │
+5 │ ╭     struct T {
+6 │ │         f: S
+7 │ │     }
+  │ ╰─────^
+  │
+  = `simple_recursion::T` contains field `f: simple_recursion::S`...
+  = `simple_recursion::S` contains field `f: T`, which forms a loop.
+
+error: recursive definition S1
+   ┌─ tests/checking/typing/recursive_struct.move:9:2
+   │
+ 9 │ ╭     struct S1 {
+10 │ │         f: S2
+11 │ │     }
+   │ ╰─────^
+   │
+   = `simple_recursion::S1` contains field `f: simple_recursion::S2`...
+   = `simple_recursion::S2` contains field `f: simple_recursion::S3`...
+   = `simple_recursion::S3` contains field `f: S1`, which forms a loop.
+
+error: recursive definition S2
+   ┌─ tests/checking/typing/recursive_struct.move:13:2
+   │
+13 │ ╭     struct S2 {
+14 │ │         f: S3
+15 │ │     }
+   │ ╰─────^
+   │
+   = `simple_recursion::S2` contains field `f: simple_recursion::S3`...
+   = `simple_recursion::S3` contains field `f: simple_recursion::S1`...
+   = `simple_recursion::S1` contains field `f: S2`, which forms a loop.
+
+error: recursive definition S3
+   ┌─ tests/checking/typing/recursive_struct.move:17:2
+   │
+17 │ ╭     struct S3 {
+18 │ │         f: S1
+19 │ │     }
+   │ ╰─────^
+   │
+   = `simple_recursion::S3` contains field `f: simple_recursion::S1`...
+   = `simple_recursion::S1` contains field `f: simple_recursion::S2`...
+   = `simple_recursion::S2` contains field `f: S3`, which forms a loop.
+
+error: recursive definition S4
+   ┌─ tests/checking/typing/recursive_struct.move:21:2
+   │
+21 │ ╭     struct S4<T> {
+22 │ │         f: S4<bool>
+23 │ │     }
+   │ ╰─────^
+   │
+   = `simple_recursion::S4<T>` contains field `f: S4`, which forms a loop.
+
+error: recursive definition S5
+   ┌─ tests/checking/typing/recursive_struct.move:25:2
+   │
+25 │ ╭     struct S5 {
+26 │ │         f: S5
+27 │ │     }
+   │ ╰─────^
+   │
+   = `simple_recursion::S5` contains field `f: S5`, which forms a loop.
+
+error: recursive definition S
+   ┌─ tests/checking/typing/recursive_struct.move:31:2
+   │
+31 │ ╭     struct S {
+32 │ │         f: G<S>
+33 │ │     }
+   │ ╰─────^
+   │
+   = `type_param::S` contains field `f: type_param::G<type_param::S>`...
```

Comment:
Why is this not displayed correctly?

Reply:
I think Teng already has a fix for this. Just remember to come back to here once his fix landed, can yours land in the meantime